client/dashboard/sites/features.ts

```diff
@@ -47,12 +47,21 @@ export const canRestorePlanSoftware = ( site: Site ) => site.is_wpcom_atomic;
 export const canDuplicateSite = ( site: Site ) => hasPlanFeature( site, DotcomFeatures.COPY_SITE );
 
 export const canTransferSite = ( site: Site, user: User ) => {
-	// TODO: The following types of the site are not allowed to transfer the ownership:
+	// The following types of the site are not allowed to transfer the ownership:
 	// * NonAtomicJetpackSite
 	// * P2 Hub
 	// * WP For Teams
 	// * VIP Site
 	// * Staging site
 	// We may need to handle this via endpoint somewhere. See canCurrentUserStartSiteOwnerTransfer.
-	return site.site_owner === user.ID;
+	const isAllowedSiteType = ! (
+		( site.jetpack && site.is_wpcom_atomic ) ||
```

Comment:
I think we're missing this exclamation mark.

```suggestion
		( site.jetpack && ! site.is_wpcom_atomic ) ||
```

I can still see the Transfer button for my jurassic.ninja site.

Reply:
Resolved by https://github.com/Automattic/wp-calypso/pull/104235/commits/ceb1d59bc016c7966a6d5ea1df246f4b1b49d198.